docs/connecting/authentication.md

```diff
@@ -0,0 +1,42 @@
+When a client connection is established Supavisor needs to verify the credentials of the connection.
+
+Credential verificiation is done either via `user` records or an `auth_query`.
+
+## Tenant User Record
+
+If no `auth_query` exists on the `tenant` record credentials will be looked up from a `user` and verified against the client connection string credentials.
+
+## Authentication Query
+
+If an `auth_query` exists on the `tenant` this query is used to query the tenant database and return results with credentials.
+
+Create a function to return a username and password for a user:
```

Comment:
perhaps, for a quick review, it's worth mentioning that it's possible to use just the following:

```sql
SELECT rolname, rolpassword FROM pg_authid WHERE rolname=$1
```